Docs/update release notes (#777)
codacy/chart: RELEASE.md

-   [ ] 4.  Inform all stakeholders outside the engineering team that a new release proccess is being started (tag @here in the Slack channel [#sh_releases](https://codacy.slack.com/channels/sh_releases)).

## 2. Before starting the release

Before starting the self-hosted release process, the Release Manager should ensure that all the features that should be introduced for this release were successfully deployed and passed all CI workflows. Also, it is recommended to check the Jira tickets to see if there are any red flags that might hinder the release process or cause any delays down the line.

The Release Manager should also confirm that the changes were properly tested before proceeding with the release. If the Release Manager finds any problems during this process, the Squads involved must be informed to find possible solutions to fix the issues before the release goes forward.

> 

Reviewer: It's great that you're giving explicit visibility to all these validations and checks @MiguelMarcelino! :muscle: Typically, these go a bit under the radar and as you mention it can cause trouble further down the line.

However, I think these validations should even be performed at an earlier stage, more specifically this is what should be validated when we mention "getting a status from each team" and "confirming if the release can go ahead" on [lines #L37-L39 above](https://github.com/codacy/chart/pull/777/files#diff-2b1b69303b927a484e02c7fad9fc87d0d3ff0dc22ae1da0ecd0dc935d922a23cR37-R39).

    git tag -f 'x.x.x-RC-<n>' && git push origin refs/tags/x.x.x-RC-<n> && git push --force-with-lease
    ```

## 4. Testing and stabilizing the release

The Release Manager must generate the release notes and announce to stakeholders the release candidate:

-   [ ] 1. Generate release notes

>     You will now have to generate the release notes for the new self-hosted release. To do so, you will need to first meet all the necessary requirements specified in the [release notes README file](https://github.com/codacy/release-notes-tools/blob/master/README.md#requirements). Please ensure that all the authentication tokens are properly setup before continuing.
>     Furthermore, you will also have to ensure that you have GitHub CLI installed.
> 
>     After ensuring that you have everything set-up properly, you can now run the makefile target `get_release_notes` to automatically generate the release notes for the new version:

Reviewer 2: We can simplify this step since the instructions you mention are described in the release notes README you link to. It's best to completely defer to that file and have only one source of truth.

```suggestion
    Make sure that you meet all the requirements specified in the [release notes README file](https://github.com/codacy/release-notes-tools/blob/master/README.md#requirements) and then run the makefile target `get_release_notes` to automatically generate the release notes for the new version:
```

Reviewer: I agree with @nicklem, let's avoid duplicating the requirements here so everything is always centralized and up to date on the codacy/release-notes-tools README.

Also, just to explain why this step didn't provide much more context before, my original idea was that the output of running the Makefile target should guide the Release Manager through fixing any setup issues and point them to the requirements on the README. But I do agree that it makes more sense to leave things explicit here. :+1:

### Approval by the Engineering teams

> -   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.

Author: I see tons of AI-suggested solutions here. Should we accept them?

> -   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.

Reviewer 2: If I understand correctly, the fixes are actually suggested by `markdownlint` (we can do this only for JS, TS, and markdown so far, see [this table](https://docs.codacy.com/getting-started/supported-languages-and-tools/#supported-languages-and-tools)), with the instructions enriched by AI.

You can ignore these suggestions; the tool checks all edited lines and the file is very likely full of such minor issues.

> -   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.

Reviewer: The issue is that the markdownlint rules for this repository aren't defined properly (for instance, someone may have applied a test coding standard to repository or something like that and the rules are now broken). Let's just ignore them at this point. :+1:

> -   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.

Reviewer: Here we can link directly to the section about exploratory tests:

```suggestion
-   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m#a649c63d) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.
```

[Read more about the release notes process](https://codacy.slite.com/app/docs/X1IGdfa0vr1Rg4) on Slite.

### Approval by the Release Manager

If there are fixes for bugs identified during the testing stage, you must [patch the current release candidate](#patching-a-release) and restart the testing and stabilization stage with the new release candidate version.

When all stakeholders have approved the release candidate, give a go-ahead of the release by clicking the Manual Approval step for QA and for the Solutions Engineers in the CircleCI workflow of your release candidate branch. After this Manual Approval on CircleCI the workflow will promote the release candidate to the [stable](https://charts.codacy.com/stable/api/charts) channel.

## 5. Launching the new release

The Release Manager must ensure that we have a stable release candidate and that both the QA team and relevant stakeholders have approved the release.

Then, the Release Manager releases and announces the new version:

-   [ ] 1.  If all is good give a public OK to the release

> -   [ ] 2.  Tag Codacy's [Analysis CLI tool](https://github.com/codacy/codacy-analysis-cli) and [CLI Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter) with the version of the release being done.

Reviewer 2: Tweak to match the official name of the CLI

```suggestion
-   [ ] 2.  Tag the [Codacy Analysis CLI tool](https://github.com/codacy/codacy-analysis-cli) and [CLI Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter) with the version of the release being done.
```

Reviewer: Yes, but even better would be to use the "official" names of both the Codacy Analysis CLI and Codacy Coverage Reporter. :nerd_face: 

```suggestion
-   [ ] 2.  Tag the [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli) and [Codacy Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter) CLI tools with the version of the release being done.
```

         git checkout x.x.x
         ```

>     3.  For the CLI Coverage Reporter, you must update the `SELF_HOSTED_CODACY_REPORTER_VERSION` flag on the `get.sh` script found in the project's root. This should reflect the new version that is intended to be used in the new self-hosted release. You do not have to do this for the Analysis CLI tool.

Reviewer 2: Tweak for conciseness

```suggestion
    3.  For the CLI Coverage Reporter, you must update the `SELF_HOSTED_CODACY_REPORTER_VERSION` flag on the `get.sh` script found in the project's root. This should match the version used by the new self-hosted release. You do not have to do this for the Analysis CLI tool.
```

Reviewer: As above, let's use the "official" names:

```suggestion
    3.  For the Codacy Coverage Reporter, you must update the `SELF_HOSTED_CODACY_REPORTER_VERSION` flag on the `get.sh` script found in the project's root. This should reflect the new version that is intended to be used in the new self-hosted release. You do not have to do this for the Codacy Analysis CLI.
```
